### DISCUSSION TITLE
gitbook cli dev server fails.

### DISCUSSION THREAD

**Discussion by ghost** [GitbookIO/community#308] [0. General bugs and questions] [answered]
Running `Node.js` version `v20.5.1` in a virtual environment. `Node.js` installed using `pip install nodeenv && nodeenv -p`. 

I have followed the instructions in the [quickstart](https://developer.gitbook.com/cli/quickstart), followed by the steps in the [cli/reference](https://developer.gitbook.com/cli/reference).

I have created a space on `gitbook` to test integrations.

When I run `gitbook dev <space id>` I get the following error:

```nodejs
\ Starting dev server...
▲ [WARNING] Cannot find base config file "@gitbook/tsconfig/integration.json" [tsconfig.json]

    tsconfig.json:2:15:
      2 │     "extends": "@gitbook/tsconfig/integration.json",
        ╵                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

X [ERROR] Could not resolve "@gitbook/runtime"

    src/index.tsx:6:9:
      6 │   } from "@gitbook/runtime";
        ╵          ~~~~~~~~~~~~~~~~~~

  You can mark the path "@gitbook/runtime" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@gitbook/runtime/jsx-runtime"

    src/index.tsx:46:10:
      46 │           <button label={element.state.message} onPress={{ action: "click" }} />
         ╵           ^

  You can mark the path "@gitbook/runtime/jsx-runtime" as external to exclude it from the bundle,
  which will remove this error.

Build failed with 2 errors:
src/index.tsx:6:9: ERROR: Could not resolve "@gitbook/runtime"
src/index.tsx:46:10: ERROR: Could not resolve "@gitbook/runtime/jsx-runtime"
```

Noticed the link to the [development section](https://developer.gitbook.com/getting-started/development) (it would be good if the links were a different color than the text in your docs.)

Apparently I need to publish the app before I can do any development.

`gitbook publish` gives this error:

```nodejs
▲ [WARNING] Cannot find base config file "@gitbook/tsconfig/integration.json" [tsconfig.json]

    tsconfig.json:2:15:
      2 │     "extends": "@gitbook/tsconfig/integration.json",
        ╵                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

X [ERROR] Could not resolve "@gitbook/runtime"

    src/index.tsx:6:9:
      6 │   } from "@gitbook/runtime";
        ╵          ~~~~~~~~~~~~~~~~~~

  You can mark the path "@gitbook/runtime" as external to exclude it from the bundle, which will
  remove this error.

X [ERROR] Could not resolve "@gitbook/runtime/jsx-runtime"

    src/index.tsx:46:10:
      46 │           <button label={element.state.message} onPress={{ action: "click" }} />
         ╵           ^

  You can mark the path "@gitbook/runtime/jsx-runtime" as external to exclude it from the bundle,
  which will remove this error.

Build failed with 2 errors:
src/index.tsx:6:9: ERROR: Could not resolve "@gitbook/runtime"
src/index.tsx:46:10: ERROR: Could not resolve "@gitbook/runtime/jsx-runtime"

```
... which appears very similar to the error I got previously.

Any suggestions?

**Comment by addisonschultz** [comment]
Hey @dougscc - Can you run `gitbook --version` and let me know what version you're using?

**Comment by ghost** [comment]
@addisonschultz Latest, according to [npm](https://www.npmjs.com/package/@gitbook/cli).

```bash
$ gitbook --version
0.10.0
```

**Comment by addisonschultz** [comment]
In the integration you're trying to publish - have you installed dependencies? `npm install`

Also, did you start your project using the gitbook cli as well? `gitbook new` - any issues running that command?

**Reply by ghost** [reply]
@addisonschultz I did run `gitbook new` to create the project.

I only ran what was in the instructions listed in the parent. I did not run `npm install` on it's own, as I foolishly expected `npm install @gitbook/cli -g` to install any dependencies it requires. I suppose that makes sense, as the cli creates a skeleton app and cannot know what the app will need in the future.

I just ran `npm install` as you suggested and got:

```bash
$ npm install

added 369 packages, and audited 370 packages in 47s

126 packages are looking for funding
  run `npm fund` for details

3 high severity vulnerabilities

To address all issues, run:
  npm audit fix

Run `npm audit` for details.

```

Running `gitbook publish` resulted in:

```bash
$ gitbook publish
🚀 Integration "cc-cross-space-search" published
👉 https://app.gitbook.com/o/cD5MHpLFFxCFjzHRcexd/integrations/cc-cross-space-search

```

1. Do I need to run `npm audit fix` to attempt to patch the vulnerabilities in `eta`?
2. Is there a way to request the addition of `npm install` to the instructions?

**Comment by ghost** [comment]
@addisonschultz unfortunately it does not appear to have helped `gitbook dev`

```bash
$ gitbook dev <space id>
\ Starting dev server...
node:events:492
      throw er; // Unhandled 'error' event
      ^

Error: spawn C:\Users\ShawhanD\Projects\gitbook-cli\venv\Scripts\node_modules\@g
itbook\cli\dist\cloudflared ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:2
1)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:2
1) {
  errno: -4058,
  code: 'ENOENT',
  syscall: 'spawn C:\\Users\\ShawhanD\\Projects\\gitbook-cli\\venv\\Scripts\\nod
e_modules\\@gitbook\\cli\\dist\\cloudflared',
  path: 'C:\\Users\\ShawhanD\\Projects\\gitbook-cli\\venv\\Scripts\\node_modules
\\@gitbook\\cli\\dist\\cloudflared',
  spawnargs: [ 'tunnel', '--no-autoupdate', '--url', 'http://localhost:51959' ]
}

Node.js v20.5.1

```

**Reply by ghost** [reply]
I can run `venv/Scripts/node_modules/@gitbook/cli/dist/cloudflared tunnel --no-autoupdate --url http://localhost:54204` from the command line and start `cloudflared`. I'm not sure what the issue with `spawn` not being able to start it from the script. I have tried starting the daemon, then running `gitbook dev <spaceid>` but no luck, yet.

**Reply by ghost** [reply]
@addisonschultz I am attempting to do all of these steps in Windows 10. I can find no direct mention of this particular problem, but there do seem to be several issues around `cloudflared` that are windows adjacent.

**Reply by addisonschultz** [reply, marked as answer]
Thanks for all the debugging - I've been looking into this with one of our engineers and seems like it's isolated to Windows machines at the moment. We've filed a bug and will have to triage this a bit more before we can fix it. 

Are you able to use a virtual machine / OS in the meantime? Apologies for the inconvenience

**Reply by ghost** [reply]
Not sure how much debugging I did. More like confused whining, but I'm inclined to agree with you: Windows problem. 😕 

I tried it on my home machine (typical Ubuntu setup) and it all worked without a hitch. Now that I'm not all stressed out about the dev environment, I can spend that energy reading all this typescript documentation (which is *vastly* better than it was the last time I looked at it in, like , 2016, not long after ECMA 6 ... Ruuuuuuussssty!).

If you all find a workaround for the `cloudflared` issue, I'll be glad to hear it. I'll go ahead and close this one, now.
